src/profile/profile.controller.ts

```diff
@@ -1,49 +1,45 @@
 import {
   Controller,
   Get,
-  Post,
   Body,
   Patch,
   Param,
-  Delete,
   UseGuards,
+  Session,
 } from '@nestjs/common';
 import { ProfileService } from './profile.service';
-import { CreateProfileDto } from './dto/create-profile.dto';
 import { UpdateProfileDto } from './dto/update-profile.dto';
 import { LoggedInGuard } from '@src/auth/guards/logged-in.guard';
-import { FindOneParams } from '@src/profile/dto/find-one.params';
+import { IdParams } from '@src/common/dto/id.params';
+import { ProfileResponseDto } from '@src/profile/dto/profile-response.dto';
 
 @UseGuards(LoggedInGuard)
 @Controller('profile')
 export class ProfileController {
   constructor(private readonly profileService: ProfileService) {}
 
-  @Post()
-  public create(@Body() createProfileDto: CreateProfileDto) {
-    return this.profileService.create(createProfileDto);
+  @Get(':id')
+  public findOneById(
+    @Param() { id }: IdParams
+  ): Promise<ProfileResponseDto> {
+    return this.profileService.findOneById(+id);
```

Comment:
Use @Transform method from library "class-transformer" to make string parameter to number